### PR TITLE
Donate 'feign-mock' to 'feign'

### PR DESCRIPTION
**Pull request by velo**
Fixes https://github.com/OpenFeign/feign/issues/687

### CODE REVIEW

File: mock/README.asciidoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# feign-mock
+
+image:https://travis-ci.org/velo/feign-mock.svg?branch=master["Build Status", link="https://travis-ci.org/velo/feign-mock"]
+image:https://coveralls.io/repos/github/velo/feign-mock/badge.svg?branch=master["Coverage Status", link="https://coveralls.io/github/velo/feign-mock?branch=master"]
+image:https://maven-badges.herokuapp.com/maven-central/com.marvinformatics.feign/feign-mock/badge.svg["Maven Central", link="https://maven-badges.herokuapp.com/maven-central/com.marvinformatics.feign/feign-mock/"]
+image:https://img.shields.io/github/issues/velo/feign-mock.svg["Issues", link="https://github.com/velo/feign-mock/issues"]
+image:https://img.shields.io/github/forks/velo/feign-mock.svg["Forks", link="https://github.com/velo/feign-mock/network"]
+image:https://img.shields.io/github/stars/velo/feign-mock.svg["Stars", link="https://github.com/velo/feign-mock/stargazers"]
+
+An easy way to test https://github.com/Netflix/feign. Since feign stores most of the logic in annotations, this helps to check if the annotations are correct.
```
Comment (rage-shadowman) [inline mock/README.asciidoc]: Should this reference "Netflix/feign"? Or "OpenFeign/feign"?

File: mock/README.asciidoc [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+The original article is available https://velo.github.io/2016/06/05/Testing-feign-clients.html[here]
+
+If mocking feign clients is easy, testing the logic written in annotations is not!
+
+To check if you are parsing the request/response properly, the only way is firing a real request. Well, that doesn't seem to be a good path to unit (or even integration) test remote services. Any IO change will affect test stability.
+
+That is why I created https://github.com/velo/feign-mock[feign-mock].
```
Comment (rage-shadowman) [inline mock/README.asciidoc]: This should probably mention something like "later merged into [here]"?

File: mock/README.asciidoc [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# feign-mock
+
+image:https://travis-ci.org/velo/feign-mock.svg?branch=master["Build Status", link="https://travis-ci.org/velo/feign-mock"]
```
Comment (spencergibb) [inline mock/README.asciidoc]: These won't be needed when merged

**Comment by velo** [comment]
Done, anything else folks?

**Comment by PaulBoutes** [comment]
LGTM

File: mock/README.asciidoc [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    assertThat(result, nullValue());
+    mockClient.verifyOne(HttpMethod.PATCH, "/repos/velo/feign-mock/contributors");
+  }
+```
+
+This simple test returns no content and verifies that the URL was truly invoked.
+
+On the mocked client, you can include all URLs and methods you want to mock.
+
+For more comprehensive examples take a look at https://github.com/velo/feign-mock/blob/master/src/test/java/feign/mock/MockClientTest.java[MockClientTest].
```
Comment (rage-shadowman) [inline mock/README.asciidoc]: Is this intentionally pointing to a file from the original repo location?

File: mock/src/main/java/feign/mock/RequestKey.java [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+            if (other.url != null) return false;
+        } else if (!url.equals(other.url)) return false;
+        return true;
+    }
+
+    public boolean equalsExtended(Object obj) {
+        if (equals(obj)) {
+            RequestKey other = (RequestKey) obj;
+            boolean headersEqual = other.headers == null || headers == null || headers.equals(other.headers);
+            boolean charsetEqual = other.charset == null || charset == null || charset.equals(other.charset);
```
Comment (rage-shadowman) [inline mock/src/main/java/feign/mock/RequestKey.java]: Does this intentionally consider `other.chaset == "UTF8"` and `charset == null` to be equal?

**Pull request merged**